### ISSUE TITLE
paymentOptionSelected not being emitted

### ISSUE THREAD

**Issue by kanedaki** [braintree/braintree-web-drop-in#272]
### General information

* SDK/Library version: 3
* Environment: sandbox
* Browser and OS:  Chrome Version 60.0.3112.113 (Official Build) (64-bit) on MacOS 10.11

### Issue description
```
        dropin
          .create({ authorization, container: '#braintree-button' })
          .then(dropinInstance => {
            dropinInstance.on('paymentOptionSelected', function() {
                console.log('paymentOptionSelected')
            })
            dropinInstance.on('paymentMethodRequestable', function() {
                console.log('paymentMethodRequestable')
            })
            dropinInstance.on('noPaymentMethodRequestable', function() {
              console.log('noPaymentMethodRequestable')
            })
            return dropinInstance
          })
```

I've a multiple payment methods dropin configured. Whenever I add a new payment method,
paymentMethodRequestable and noPaymentMethodRequestable, are being fired correctly, but when I change from one already created method to another, there is no event being fired.
![screen shot 2017-08-31 at 5 07 57 pm](https://user-images.githubusercontent.com/565038/29930575-faf7e99c-8e6e-11e7-8b2c-2b84e7ca04c6.png)
![screen shot 2017-08-31 at 5 08 38 pm](https://user-images.githubusercontent.com/565038/29930601-115f43ba-8e6f-11e7-8894-28f86040ea39.png)

In other words, when I click on the first option of the second image, and the green mark change from the second option to the first, no event is being fired. As far as I understood, `paymentOptionSelected` is fired when selecting a different option, right? If this is not the case, how can I detect a change?

Thanks in advance

**Comment by kzurawel** [comment]
There is currently no way to detect changes in payment method within the same type of payment (e.g. from one credit card to another). The `paymenOptionSelected` event occurs when the user switches between _different types_ of payment (e.g. from credit card to PayPal). We will update our docs to make this more clear.

Would you mind telling us more about your use case? There might be another way to achieve what you're after.

Thanks!

**Comment by crookedneighbor** [comment]
In addition, the paymentOptionSelected event only fires when a payment option is selected. Not when selecting a previously vaulted/stored payment method.

**Comment by crookedneighbor** [comment]
We've updated the docs with #273 to reflect that. Going to close this now, but feel free to leave a comment if you have a use case for having an event when a payment method is selected.

**Comment by kanedaki** [comment]
Hi,

Thanks for your quick answer @crookedneighbor . I would like to expose my use case. Probably there is already a way to detect what I need.

I've a customer with two credit cards stored on the vault. So, every time I show the dropin, there are two payment methods. Let's say the client fill in another one. I call the requestPaymentMethod, I get a nonce. But I did not store the nonce, I use it to get a paymentMethodToken which I can use on several payments on the future. This is a common use case. But if client, instead of creating a new one, wants to select one of the previous payment methods, how to automatically call requestPaymentMethod, when I user changes between them

Thanks in advance

**Comment by crookedneighbor** [comment]
I've opened a PR to accomplish what you are asking for: https://github.com/braintree/braintree-web-drop-in/pull/274

In the mean time, you can also use [clearSelectedPaymentMethod](https://braintree.github.io/braintree-web-drop-in/docs/current/Dropin.html#clearSelectedPaymentMethod) after Drop-in has been created to clear off the pre-selected payment method. 

This will have a visual artifact as the payment method is cleared, so you may want to hide the drop-in until after you've cleared it.

Ultimately, the PR I opened will be a better solution.

**Comment by crookedneighbor** [comment]
Closed with https://github.com/braintree/braintree-web-drop-in/pull/274

Will be available in the next release

**Comment by idontusenumbers** [comment]
Any update on this? It's hard to track down what releases there are and what the latest versions of the libraries are.

**Comment by crookedneighbor** [comment]
This was released as part of 1.8.0. See https://github.com/braintree/braintree-web-drop-in/blob/master/CHANGELOG.md#180

**Comment by rw3iss** [comment]
Do I understand this correctly?
There is no event that fires if a user changes between different credit cards (same payment "option"/type)? 
If I clear the selection a user currently selects a credit card, using instance.clearSelectedPaymentMethod(), and then the user selects one of the saved credit cards, it will fire the paymentMethodRequestable event, and I can grab the nonce for that card, but then if the user changes the selected credit cards, still no event fires, and I can't change the nonce.

Why is there no event for this situation? Seems like a huge hole in the design.

It may be hackish, but we could possibly do:
-when a user selected a new payment (with nothing previously selected), and the paymentMethodRequestable event fires, we can grab the none then, store it in local state, and then use instance.clearSelectedPaymentMethod(), and customise our own UI a bit to show that a method has been selected, and only send the nonce to our server. If the user changes the selection, we would do the same thing...

Still, there should be an event that fires when any new payment option (regardless of type) changes!
